src/script/components/fetch-token.mjs

```diff
@@ -0,0 +1,77 @@
+const givenToken = localStorage.getItem("accessToken");
+
+const getPostsOption = {
+  method: "GET",
+  headers: {
+    "Content-Type": "application/json",
+    Authorization: `Bearer ${givenToken}`,
+  },
+};
+
+async function getPosts(url) {
+  getPostsOption;
+  const response = await fetch(url, getPostsOption);
+  const posts = await response.json();
+  if (response.ok) {
+    return posts;
+  } else {
+    throw new Error("Could not get posts!");
+  }
+}
+
+async function createPost(url, data) {
+  const postData = {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${givenToken}`,
+    },
+    body: JSON.stringify(data),
+  };
+  const response = await fetch(url, postData);
+  const json = await response.json();
+
+  if (response.ok) {
+    return json;
+  } else {
+    throw new Error("Could not create a new post!");
+  }
+}
+
+async function deletePost(url) {
```

Comment:
This implementation looks good, but it is not connected to the UI. I cannot delete a post. It is one of the requirements, and you need to fix it before you can pass. The same goes for updating functionality.